test/test-barrier.c

```diff
@@ -27,20 +27,22 @@
 
 typedef struct {
   uv_barrier_t barrier;
-  int delay;
-  volatile int posted;
-  int main_barrier_wait_rval;
-  int worker_barrier_wait_rval;
+  unsigned delay;
+  unsigned niter;
+  unsigned main_barrier_wait_rval;
+  unsigned worker_barrier_wait_rval;
 } worker_config;
 
 
 static void worker(void* arg) {
   worker_config* c = arg;
+  unsigned i;
 
   if (c->delay)
     uv_sleep(c->delay);
 
-  c->worker_barrier_wait_rval = uv_barrier_wait(&c->barrier);
+  for (i = 0; i <= c->niter; i++)
+      c->worker_barrier_wait_rval += uv_barrier_wait(&c->barrier);
```

Comment:
```suggestion
  for (i = 0; i <= c->niter; i++)
    c->worker_barrier_wait_rval += uv_barrier_wait(&c->barrier);
```
Also applies to your other for loops. :-)

And maybe `i < c->niter`? Then you don't have to check `wc.niter + 1` below, just `wc.niter`. Less surprising.

Reply:
Then I needed to set it everywhere, vs making 0 the default value. But your call. Either is easy.

Reply:
Fair enough. Either way is fine.